src/io/flutter/actions/FlutterSdkAction.java

```diff
@@ -120,5 +82,13 @@ public void actionPerformed(AnActionEvent event) {
     }
   }
 
-  public abstract void perform(@NotNull FlutterSdk sdk, @NotNull Project project, AnActionEvent event) throws ExecutionException;
+  public final void perform(@NotNull FlutterSdk sdk, @NotNull Project project, @Nullable AnActionEvent event, boolean logAction)
+    throws ExecutionException {
+    if (logAction) {
+      sendActionEvent();
+    }
+    perform(sdk, project, event);
```

Comment:
I'm going for a template method though. I don't want implementers to see (or care) about the extra `logAction` bit.  Or maybe I'm misunderstanding?